src/test/suite/extensionFunctions.test.ts

```diff
@@ -0,0 +1,44 @@
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import {extensionFunctions} from '../../extensionFunctions';
+
+suite('Extension Functions Test Suite', () => {
+
+	test('send text to terminal', async function() {
+		const testTerminalName = 'testTerminal';
+		var terminal : vscode.Terminal | undefined = extensionFunctions.findTerminal(testTerminalName);
+		assert.equal(terminal, undefined);
+
+		await extensionFunctions.sendTerminalText(testTerminalName, "testText");
+		terminal = extensionFunctions.findTerminal(testTerminalName);
+		assert.notEqual(terminal, undefined);
+
+		// open to ideas on how to check to see that a message was actually committed to the terminal
+		// but at least we can check to see if the terminal was created as part of the sendTerminalText call
+	});
+
+	test('send ctrl+c to terminal', async function() {
+		const testTerminalName = 'testTerminalCtrlC';
+		var terminalC : vscode.Terminal | undefined = extensionFunctions.findTerminal(testTerminalName);
+		assert.equal(terminalC, undefined);
+
+		// open to ideas on how to check to see that ctrl+c was actually committed to the terminal
+		await extensionFunctions.sendTerminalCtrlC(testTerminalName);
+
+		// terminal should not have been created as part of the method call
+		terminalC = extensionFunctions.findTerminal(testTerminalName);
+		assert.equal(terminalC, undefined);
+
+		await extensionFunctions.startTerminal(testTerminalName);
+		// terminal should have been created as part of the method call
+		terminalC = extensionFunctions.findTerminal(testTerminalName);
+		assert.notEqual(terminalC, undefined);
+
+		// again, open to ideas on how to check to see if the terminal text was sent to the active terminal
```

Comment:
we can use "side-effects", like the text sent to the terminal is to create a file somehwere and then we check that the file is created. But it seems a bit too effort for the value. I prefer the idea to wait for listener outputchannel content pointed by Lars to be in VS Code APi.

Reply:
Sounds good. I'll create an issue so we can update the tests when we see the new functionality in an upcoming build.

Reply:
oh wait here we are on the Ctrl, so the terminal shoudl have been deleted, right? so would say current test is enough

Reply:
Terminal does not get deleted - that's a different issue. The terminal just halts any long-running process.

Reply:
oh my bad I misunderstood, so I guess we will wait for the listener APi to be available

Reply:
https://github.com/redhat-developer/vscode-didact/issues/49 will add a new command to close the terminal